Error messages where:
```
-- Found CUDAToolkit: /home/axel/micromamba/envs/warpx-gpu-mpich-dev/targets/x86_64-linux/include (found version "12.9.86")
CMake Error at build/_deps/fetchedamrex-src/Tools/CMake/AMReXUtils.cmake:229 (math):
  math cannot parse the expression: "50-real/10": syntax error, unexpected
  exp_DIVIDE (8).
Call Stack (most recent call first):
  build/_deps/fetchedamrex-src/Tools/CMake/AMReXUtils.cmake:262 (convert_cuda_archs)
  build/_deps/fetchedamrex-src/Tools/CMake/AMReXParallelBackends.cmake:113 (set_cuda_architectures)
  build/_deps/fetchedamrex-src/Src/CMakeLists.txt:40 (include)


CMake Error at build/_deps/fetchedamrex-src/Tools/CMake/AMReXUtils.cmake:230 (math):
  math cannot parse the expression: "50-real%10": syntax error, unexpected
  exp_MOD (8).
Call Stack (most recent call first):
  build/_deps/fetchedamrex-src/Tools/CMake/AMReXUtils.cmake:262 (convert_cuda_archs)
  build/_deps/fetchedamrex-src/Tools/CMake/AMReXParallelBackends.cmake:113 (set_cuda_architectures)
  build/_deps/fetchedamrex-src/Src/CMakeLists.txt:40 (include)
```

and
```
CMake Error at /home/axel/micromamba/envs/warpx-gpu-mpich-dev/share/cmake-3.30/Modules/FindCUDA/select_compute_arch.cmake:245 (message):
  Unknown CUDA Architecture Name 10.0 in CUDA_SELECT_NVCC_ARCH_FLAGS
Call Stack (most recent call first):
  build/_deps/fetchedamrex-src/Tools/CMake/AMReXUtils.cmake:274 (cuda_select_nvcc_arch_flags)
  build/_deps/fetchedamrex-src/Tools/CMake/AMReXParallelBackends.cmake:113 (set_cuda_architectures)
  build/_deps/fetchedamrex-src/Src/CMakeLists.txt:40 (include)


CMake Error at /home/axel/micromamba/envs/warpx-gpu-mpich-dev/share/cmake-3.30/Modules/FindCUDA/select_compute_arch.cmake:249 (message):
  arch_bin wasn't set for some reason
Call Stack (most recent call first):
  build/_deps/fetchedamrex-src/Tools/CMake/AMReXUtils.cmake:274 (cuda_select_nvcc_arch_flags)
  build/_deps/fetchedamrex-src/Tools/CMake/AMReXParallelBackends.cmake:113 (set_cuda_architectures)
  build/_deps/fetchedamrex-src/Src/CMakeLists.txt:40 (include)


CMake Error at /home/axel/micromamba/envs/warpx-gpu-mpich-dev/share/cmake-3.30/Modules/FindCUDA/select_compute_arch.cmake:245 (message):
  Unknown CUDA Architecture Name 10.1 in CUDA_SELECT_NVCC_ARCH_FLAGS
Call Stack (most recent call first):
  build/_deps/fetchedamrex-src/Tools/CMake/AMReXUtils.cmake:274 (cuda_select_nvcc_arch_flags)
  build/_deps/fetchedamrex-src/Tools/CMake/AMReXParallelBackends.cmake:113 (set_cuda_architectures)
  build/_deps/fetchedamrex-src/Src/CMakeLists.txt:40 (include)


CMake Error at /home/axel/micromamba/envs/warpx-gpu-mpich-dev/share/cmake-3.30/Modules/FindCUDA/select_compute_arch.cmake:249 (message):
  arch_bin wasn't set for some reason
Call Stack (most recent call first):
  build/_deps/fetchedamrex-src/Tools/CMake/AMReXUtils.cmake:274 (cuda_select_nvcc_arch_flags)
  build/_deps/fetchedamrex-src/Tools/CMake/AMReXParallelBackends.cmake:113 (set_cuda_architectures)
  build/_deps/fetchedamrex-src/Src/CMakeLists.txt:40 (include)
...
```

--- a/Tools/CMake/AMReXUtils.cmake
+++ b/Tools/CMake/AMReXUtils.cmake
@@ -222,15 +222,22 @@ endfunction ()
 function (convert_cuda_archs  _cuda_archs)
 
    foreach (_item IN LISTS ${_cuda_archs})
+      # remove -real suffixes
+      string(REGEX REPLACE "\\-real$" "" _item "${_item}")
+
       string(REGEX MATCH "\\." _has_decimal "${_item}")
       string(REGEX MATCH "[0-9]+" _is_number "${_item}")
 
       if (NOT _has_decimal AND _is_number)
          math(EXPR _int "${_item}/10" OUTPUT_FORMAT DECIMAL)
          math(EXPR _mod "${_item}%10" OUTPUT_FORMAT DECIMAL)
-         list(APPEND _tmp "${_int}.${_mod}")
+         if(_int LESS 10)  # CMake 3.30 does not support SM 10.0+ in cuda_select_nvcc_arch_flags
+             list(APPEND _tmp "${_int}.${_mod}")
+         endif()
       else ()
-         list(APPEND _tmp ${_item})
+         if(_item LESS 10)  # CMake 3.30 does not support SM 10.0+ in cuda_select_nvcc_arch_flags
+            list(APPEND _tmp ${_item})
+        endif()
       endif()
    endforeach ()
 

